File: exams/artieda/u3/Strategy/src/main/java/ec/edu/espe/strategy/model/BubbleSort.java
package ec.edu.espe.strategy.model;

public class BubbleSort implements SortingStrategy {

    @Override
    public int[] sort(int[] data) {
        if (data == null) {
            return new int[0];
        }
        int[] sorted = data.clone();
        int n = sorted.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (sorted[j] > sorted[j + 1]) {
                    int temp = sorted[j];
                    sorted[j] = sorted[j + 1];
                    sorted[j + 1] = temp;
                }
            }
        }
        return sorted;
    }
}

File: exams/tipan/u3/Exam U3/StrategyJS/model/BubbleSort.js
const SortingStrategy = require("./SortingStrategy");

class BubbleSort extends SortingStrategy {

    sort(array) {
        let result = [...array];

        for (let i = 0; i < result.length - 1; i++) {
            for (let j = 0; j < result.length - i - 1; j++) {
                if (result[j] > result[j + 1]) {
                    let temp = result[j];
                    result[j] = result[j + 1];
                    result[j + 1] = temp;
                }
            }
        }

        return result;
    }
}

module.exports = BubbleSort;